src/stim/stabilizers/tableau.cc

```diff
@@ -379,6 +386,31 @@ bool Tableau::satisfies_invariants() const {
     return true;
 }
 
+bool Tableau::is_conjugation_by_pauli() {
+    simd_bit_table<MAX_BITWORD_WIDTH> scratch_matrix =
+        simd_bit_table<MAX_BITWORD_WIDTH>::identity(num_qubits);
+
+    if ( xs.xt != scratch_matrix || zs.zt != scratch_matrix ) {
+        return false;
+    }
+    scratch_matrix.clear();
+    if ( xs.zt != scratch_matrix || zs.xt != scratch_matrix) {
+        return false;
+    }
+    return true;
+}
+
+PauliString Tableau::to_pauli_string() {
+    if ( ! is_conjugation_by_pauli() ) {
+        throw std::invalid_argument("tableau is not conjugation by a pauli");
```

Comment:
`The tableau isn't equivalent to a Pauli product.`

Reply:
done